ui/app/models/pki/issuer.js

```diff
@@ -1,49 +1,69 @@
-import PkiCertificateBaseModel from './certificate/base';
-import { attr } from '@ember-data/model';
+import Model, { attr } from '@ember-data/model';
 import { withFormFields } from 'vault/decorators/model-form-fields';
 import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
+import { service } from '@ember/service';
 
 const issuerUrls = ['issuingCertificates', 'crlDistributionPoints', 'ocspServers'];
-@withFormFields(
-  ['issuerName', 'leafNotAfterBehavior', 'usage', 'manualChain', ...issuerUrls],
-  [
-    {
-      default: [
-        'certificate',
-        'caChain',
-        'commonName',
-        'issuerName',
-        'issuerId',
-        'serialNumber',
-        'keyId',
-        'uriSans',
-        'ipSans',
-        'notValidBefore',
-        'notValidAfter',
-      ],
-    },
-    { 'Issuer URLs': issuerUrls },
-  ]
-)
-export default class PkiIssuerModel extends PkiCertificateBaseModel {
-  // there are too many differences between what openAPI returns and the designs for the update form
-  // manually defining the attrs instead with the correct meta data
+const inputFields = [
+  'issuerName',
+  'leafNotAfterBehavior',
+  'usage',
+  'manualChain',
+  'revocationSignatureAlgorithm',
+  ...issuerUrls,
+];
+const displayFields = [
+  {
+    default: [
+      'certificate',
+      'caChain',
+      'commonName',
+      'issuerName',
+      'issuerId',
+      'serialNumber',
+      'keyId',
+      'altNames',
+      'uriSans',
+      'ipSans',
+      'otherSans',
+      'notValidBefore',
+      'notValidAfter',
+    ],
+  },
+  { 'Issuer URLs': issuerUrls },
+];
+@withFormFields(inputFields, displayFields)
+export default class PkiIssuerModel extends Model {
+  @service secretMountPath;
+  // TODO use openAPI after removing route extension (see pki/roles route for example)
```

Comment:
Not sure what exactly `after removing route extension` means -- could you clarify?

Reply:
We can't use openApi for issuers until we refactor like we did for roles here https://github.com/hashicorp/vault/pull/18980

I think we've also decided to try and avoid extending routes because of this. Kianna put up a PR to change the config to a decorator instead.